zstd/seqdec.go

```diff
@@ -181,11 +181,13 @@ func (s *sequenceDecs) decode(seqs int, br *bitReader, hist []byte) error {
 			return fmt.Errorf("output (%d) bigger than max block size", size)
 		}
 		if size > cap(s.out) {
-			// Not enough size, will be extremely rarely triggered,
+			// Not enough size, which can happen under high volume block streaming conditions
 			// but could be if destination slice is too small for sync operations.
-			// We add maxBlockSize to the capacity.
-			s.out = append(s.out, make([]byte, maxBlockSize)...)
-			s.out = s.out[:len(s.out)-maxBlockSize]
+			// over-allocating here can create a large amount of GC pressure so we try to keep
+			// it as contained as possible
+			addBytes := size - len(s.out)
```

Comment:
There will likely be thousands of sequence elements, so this will lead to potentially as many allocations and copies for a huge slowdown.

```suggestion
			addBytes :=  1024 + len(s.out) >> 2
```

This will add 1K and half the size of the current output. It should achieve the same, but with far less allocations.

Reply:
ok .. got a panic using that exact logic

```
panic: runtime error: slice bounds out of range [:27621] with capacity 24182
```